File: frontend/src/microservices/advertising/objects/AdvertisementSlotAssignmentRequest.ts
// 本文件定义 advertising 模块的 `AdvertisementSlotAssignmentRequest`，用于广告投放位分配并提供 JSON 编解码。

export type AdvertisementSlotAssignmentRequest = {
  reviewerManagerId: string
  slotIndex: number
}

export const advertisementSlotAssignmentRequestFromJson = (json: string): AdvertisementSlotAssignmentRequest =>
  JSON.parse(json) as AdvertisementSlotAssignmentRequest

export const advertisementSlotAssignmentRequestToJson = (value: AdvertisementSlotAssignmentRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/AdvertisementSlotAssignmentRequest.scala
// 本文件定义 advertising 模块的 AdvertisementSlotAssignmentRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AdvertisementSlotAssignmentRequest(
    advertisementId: String,
    reviewerManagerId: String,
    slotIndex: Int
)

object AdvertisementSlotAssignmentRequest:
  given sourceEncoder: Encoder[AdvertisementSlotAssignmentRequest] = deriveEncoder
  given sourceDecoder: Decoder[AdvertisementSlotAssignmentRequest] = deriveDecoder

